### PR TITLE
Update deploy file test

### PR DESCRIPTION
**Pull request by batikanu**
Solution for issue #318 is implemented. The created namespace is deleted at the end of the execution.

### CODE REVIEW

File: src/test/backend/deploy_test.go [hunk 1]
```diff
@@ -177,6 +183,13 @@ func TestDeployAppFromFileWithValidContent(t *testing.T) {
 	if err != nil {
 		t.Errorf("Expected return value to be %#v but got %#v", nil, err)
 	}
+
+	//cleaning the created namespace
+	factory := cmdutil.NewFactory(nil)
```
Comment (bryk) [inline src/test/backend/deploy_test.go]: If the test fail for some reason or I stop it in the middle of the work, the namespace will still be there. I think that the only solution is to create an interface for the dashboard <-> kubectl boundary.  For example, you could create an interface for the line:
`_, err = kubectlResource.NewHelper(info.Client, info.Mapping).Create(info.Namespace, true, info.Object)`

In tests you'd pass a verifying callback. On prod you'd call kubectl implementation callback.

Comment (bryk) [inline src/test/backend/deploy_test.go]: Like, `createObjectFromInfo`.

Comment (batikanu) [inline src/test/backend/deploy_test.go]: Good idea! I will check it.

**Comment by batikanu** [comment]
PTAL

File: src/app/backend/deploy.go [hunk 2]
```diff
@@ -243,8 +243,20 @@ func getLabelsMap(labels []Label) map[string]string {
 	return result
 }
 
+type ObjectFromInfo interface {
+	createObjectFromInfo(info *kubectlResource.Info) error
+}
+
+type ResourceFromInfo string
```
Comment (bryk) [inline src/app/backend/deploy.go]: You dont need to do this. I.e., creating fake struct that fulfills an interface. You can just have a callback function as the argument. @floreks did this recently in PR https://github.com/kubernetes/dashboard/pull/319 , you can talk with him.

**Comment by bryk** [comment]
Looks very nice. One comment on types. PTAL

**Comment by batikanu** [comment]
I've implemented a callback function. PTAL

**Comment by bryk** [comment]
Awesome, thanks!

**Pull request merged**